app/services/slugs/update_sluggable_history.rb

```diff
@@ -0,0 +1,19 @@
+class Slugs::UpdateSluggableHistory
+  def self.call(*args)
+    new(*args).call
+  end
+
+  def initialize(slug_prefix:, resource:)
+    @slug_prefix = slug_prefix
+    @resource = resource
+  end
+
+  def call
+    @resource.slugs.active.update_all(active: false) &&
+    @resource.slugs.active.build(
```

Comment:
That's why [app/models/concerns/sluggable.rb:25](#diff-25d31258d87084585456d7181a6447e2R25) needs to be changed to `after_save`.